core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -1285,15 +1296,35 @@ public void finishUpdate(Session session, TableHandle tableHandle, Collection<Sl
             throw new TrinoException(SCHEMA_NOT_FOUND, format("Schema '%s' does not exist", schemaName));
         }
         CatalogMetadata catalogMetadata = getCatalogMetadata(session, new CatalogName(schemaName.getCatalogName()));
-        CatalogName catalogName = catalogMetadata.getConnectorIdForSchema(schemaName);
-        ConnectorMetadata metadata = catalogMetadata.getMetadataFor(catalogName);
+        if (catalogMetadata.getSecurityManagement() == SecurityManagement.SYSTEM) {
+            return Optional.of(systemSecurityMetadata.getSchemaOwner(session, schemaName));
+        }
+        else {
+            CatalogName catalogName = catalogMetadata.getConnectorIdForSchema(schemaName);
+            ConnectorMetadata metadata = catalogMetadata.getMetadataFor(catalogName);
+            ConnectorSession connectorSession = session.toConnectorSession(catalogName);
+            return metadata.getSchemaOwner(connectorSession, schemaName);
+        }
+    }
 
-        ConnectorSession connectorSession = session.toConnectorSession(catalogName);
-        return metadata.getSchemaOwner(connectorSession, schemaName);
+    @Override
+    public boolean isView(Session session, QualifiedObjectName viewName)
+    {
+        return getViewInternal(session, viewName).isPresent();
     }
 
     @Override
     public Optional<ViewDefinition> getView(Session session, QualifiedObjectName viewName)
+    {
+        Optional<ViewDefinition> view = getViewInternal(session, viewName);
+        if (view.isEmpty() || view.get().getRunAsIdentity().isEmpty() || getCatalogMetadata(session, new CatalogName(viewName.getCatalogName())).getSecurityManagement() == SecurityManagement.CONNECTOR) {
```

Comment:
Add a method `isRunAsInvoker()` to `ViewDefinition` so it's clear why we don't use the result of `getRunAsIdentity()` when it is present